.github/workflows/build-test-format.yml

```
@@ -7,22 +7,70 @@ on:
     branches: [ "main" ]
   workflow_dispatch:
 
+env:
+  RUST_FMT: nightly-2023-04-01-x86_64-unknown-linux-gnu
+  RUST_CLIPPY: 1.68
+
 jobs:
-  build-test-format:
+  rustfmt:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: Set up Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ env.RUST_FMT }}
+        override: true
+        components: rustfmt
+
+    - name: Format Rust Project
+      run: cargo fmt --manifest-path=./rust-bindings/Cargo.toml --all --check
```

```suggestion
    - name: Format Rust Project
      run: |
        rustup default ${{ env.RUST_FMT }}
        rustup component add rustfmt
        cargo fmt --manifest-path=./rust-bindings/Cargo.toml --all --check
```

I recommend avoiding actions-rs in new pipelines since it is unmaintained. I recently migrated all of concordium-base jobs away from using these actions. The base ubuntu image has everything we need already.